### PR TITLE
Remove dashboards from APM Server.

### PR DESCRIPTION
**Pull request by simitt**
From 7.0 on APM dashboards and visualisations will no longer be bundled directly with Kibana or APM Server.

depends on #https://github.com/elastic/kibana/pull/29043
depends on https://github.com/elastic/beats/pull/10277/

closes elastic/apm-server#1673

cc @sqren 

UPDATE:
Updating to the latest libbeat changes that allow to manage a commands flags, also pulled in some more changes on the kibana index pattern. This PR now depends on merging the index pattern changes in to Kibana https://github.com/elastic/kibana/pull/29501

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -72,4 +72,15 @@ func init() {
...
 	})
+	// remove dashboard from export commands
+	for _, cmd := range RootCmd.ExportCmd.Commands() {
+		if cmd.Name() == "dashboard" {
+			RootCmd.ExportCmd.RemoveCommand(cmd)
+		}
+	}
+	// remove dashboards and ml-jobs from setup commands
+	setup := RootCmd.SetupCmd
+	setup.ResetFlags()
```
Comment (jalvz) [inline cmd/root.go]: Running `./apm-server setup` prints
```Loaded index template
Loading dashboards (Kibana must be running and reachable)
Skipping loading dashboards, No directory /home/juan/go/src/github.com/elastic/apm-server/kibana/6
```
which might confuse users (looks a bit like an error, when it is not)

Also `./apm-server help` references dashboards.

Comment (simitt) [reply cmd/root.go]: I will investigate, I am sure this worked when I tried it out.

Comment (simitt) [reply cmd/root.go]: I checked again, you are right that it still shows up in the description, it was only removed in the flags listed in the help. I'll push a follow up.

Comment (simitt) [reply cmd/root.go]: depends on https://github.com/elastic/beats/pull/10277/

**Comment by simitt** [comment]
jenkins, run package tests please

**Comment by simitt** [comment]
jenkins, retest this please

**Comment by jalvz** [comment]
forgot to mention: i guess we can exclude the sync of `dev-tools/cmd/dashboards` too

**Comment by simitt** [comment]
Thanks for the thorough review @jalvz ! 

* `make dashboards` is only developer utility, so I didn't change anything, but you're right I should. I pushed a change. 

* Regarding the docs: there is a PR https://github.com/elastic/apm-server/pull/1791 where @bmorelli25 removes everything dashboard related. If something is overlooked, please note it there. I didn't update any docs. 

* There is an open PR https://github.com/elastic/beats/pull/10138/files#diff-9f2f32ca2002718c833fab82f71c64ce to remove the `--setup` flag. Afterwards I expect the setup not to show up for the beats help at all, so I didn't bother to change that. 

* I now excluded the `dev-tools/cmd/dashboards` from being synced and removed it. Great finding!

> Unrelated to this PR, just an observation i noticed: is a bit weird that if you pass any gibberish to the export command then it shows the help, but if you pass it to the setup command it is ignored

This is because the `export` cmd actually expects information what to export, eg. `./apm-server export template`, but the `setup` cmd just sets up everything if you don't pass additional information. If you add the  _gibberish_ after the full cmd (`export template abc`), it behaves the same.

**Comment by simitt** [comment]
jenkins, run package tests please

**Comment by jalvz** [comment]
Oh, sorry for the comment on the docs, i thought that PR was merged. Ignore it then.

**Pull request merged**